plugins/org.jboss.reddeer.workbench/src/org/jboss/reddeer/workbench/impl/editor/internal/BasicEditorFile.java

```diff
@@ -0,0 +1,75 @@
+/******************************************************************************* 
+ * Copyright (c) 2017 Red Hat, Inc. 
+ * Distributed under license by Red Hat, Inc. All rights reserved. 
+ * This program is made available under the terms of the 
+ * Eclipse Public License v1.0 which accompanies this distribution, 
+ * and is available at http://www.eclipse.org/legal/epl-v10.html 
+ * 
+ * Contributors: 
+ * Red Hat, Inc. - initial API and implementation 
+ ******************************************************************************/
+package org.jboss.reddeer.workbench.impl.editor.internal;
+
+import java.io.InputStream;
+
+import org.eclipse.core.resources.IFile;
+import org.eclipse.core.runtime.CoreException;
+import org.jboss.reddeer.workbench.api.EditorFile;
+import org.jboss.reddeer.workbench.exception.WorkbenchLayerException;
+
+/**
+ * Basic implementation of EditorFile which represents a file associated with an
+ * editor.
+ * 
+ * @author Andrej Podhradsky (apodhrad@redhat.com)
+ *
+ */
+public class BasicEditorFile implements EditorFile {
+
+	private IFile iFile;
+
+	/**
+	 * Constructs the a implementation of EditorFile from IFile.
+	 * 
+	 * @param iFile
+	 *            IFile
+	 */
+	public BasicEditorFile(IFile iFile) {
+		this.iFile = iFile;
+	}
+
+	@Override
+	public InputStream getInputStream() {
```

Comment:
well, it would be similar to Widgets where we return Eclipse SWT.

Nothing wrong is with Input stream. I was pointing that the method to get input stream is used only in another method of the same class. And I am not quite sure whether anybody would use that method outside of the class. If we would provide underlying resource, we provide more power to end users to use it as they want.